probe/ebpf/bpf.h

```diff
@@ -28,6 +28,28 @@
 # define memmove(dest, src, n)  __builtin_memmove((dest), (src), (n))
 #endif
 
+/*
+The memcmp() built-in had some corner cases where inlining did not take place due to an LLVM issue in the back end, and is therefore not recommended to be used until the issue is fixed.
+*/
+//#define LLVM_MEMCMP_WORKING 1
+#ifdef LLVM_MEMCMP_WORKING
+#ifndef memcmp
+# define memcmp(src1, src2, n)   __builtin_memcmp((src1), (src2), (n))
+#endif
+#else
+#define memcmp discret_memcmp
+static __inline int discret_memcmp(char *s1, char *s2, int n) {
+	int i;
+#pragma unroll
+	for(i=0;i<n;s1++,s2++,i++) {
+		if (*s1 != *s2) {
+			return 1;
```

Comment:
memcmp(2) return value is:

> RETURN VALUE
> The  memcmp() function returns an integer less than, equal to, or greater than zero if the first n bytes of s1 is found, respectively, to be less than, to match, or be greater than the first n bytes of s2.
>
> For a nonzero return value, the sign is determined by the sign of the difference between the first pair of bytes (interpreted as unsigned char) that differ in s1 and s2.
>
> If n is zero, the return value is zero.

If we ever rely on the negative/zero/positive return value (for example for using in qsort()), then this implementation of memcmp will break (because it just returns 0 or 1).